Feat: adjust fields for ECS compatibility (#19)
logstash-plugins/logstash-input-pipe: docs/index.asciidoc

want to join lines, you'll want to use the multiline codec.

[id="plugins-{type}s-{plugin}-ecs"]
==== Compatibility with the Elastic Common Schema (ECS)

This plugin adds extra fields about the event's source, and can be configured to do so

> in an {ecs-ref}[ECS-compatible] way with <<plugins-{type}s-{plugin}-ecs_compatibility>>.

Reviewer: ```suggestion
in an {ecs-ref}[ECS-compatible] way with the <<plugins-{type}s-{plugin}-ecs_compatibility>> option.
```

> This plugin adds extra fields about the event's source, and can be configured to do so
> in an {ecs-ref}[ECS-compatible] way with <<plugins-{type}s-{plugin}-ecs_compatibility>>.

Reviewer: ```suggestion
This plugin adds extra fields about the event's source. 
Configure the <<plugins-{type}s-{plugin}-ecs_compatibility>> option to ensure
that these fields are compatible with {ecs-ref}[ECS].

```

Reviewer: The link text should give users a clear idea of where the link will take them. "ECS-compatible" is not 100% clear and looks a lot like the link for "ecs_compatibility" option.  My intention is to have the "ECS" text link to "ECS," and "ecs_compatibility" link to the "ecs_compatibility" option.

Example:
[source,ruby]
-----
input {
  pipe {
    command => "echo ¡Hola!"
  }
}
-----

[id="plugins-{type}s-{plugin}-ecs_compatibility"]
===== `ecs_compatibility`

  * Value type is <<string,string>>
  * Supported values are:
    ** `disabled`: uses backwards compatible field names, such as `[host]`
    ** `v1`, `v8`: uses fields that are compatible with ECS, such as `[host][name]`

Controls this plugin's compatibility with the {ecs-ref}[Elastic Common Schema (ECS)].
See <<plugins-{type}s-{plugin}-ecs>> for detailed information.

**Sample output: ECS enabled**
[source,ruby]
-----
{
    "@timestamp"=>2021-11-16T09:18:45.306Z,
    "message" => "¡Hola!",
    "process" => {
        "command_line" => "echo '¡Hola!'"
    },
    "host" => {
        "name" => "deus-ex-machina"
    }
}
-----

**Sample output: ECS disabled**
[source,ruby]
-----
{
    "@timestamp"=>2021-11-16T09:18:45.306Z,
    "message" => "¡Hola!",
    "command" => "echo '¡Hola!'",
    "host" => "deus-ex-machina"
}

> -----

Reviewer: Thanks for adding these examples. Nice!
